primitives/line.go

```diff
@@ -11,3 +11,33 @@ func (l Line) ToVector() Vector {
 func (l Line) Reverse() Line {
 	return Line{l.P2, l.P1}
 }
+
+func (l Line) IntersectLine(l2 *Line) *Point {
+
+	a := l.ToVector()
+	b := l2.ToVector()
```

Comment:
please check Vlad's PR, function IsIntersectingSegment, it might be a better version.